src/pool/srv_pool.c

```diff
@@ -6975,10 +6974,16 @@ pool_svc_update_map_internal(struct pool_svc *svc, unsigned int opc,
 		goto out_map;
 	}
 
-	/* For SWIM exclude, don't change pool map if the pw_rf is broken or is going to be broken,
-	 * with CRIT log message to ask administrator to bring back the engine.
+	/*
+	 * For SWIM exclude:
+	 * Do not change pool map if the pw_rf is broken or is going to be broken,
+	 * With CRIT log message to ask administrator to bring back the engine.
+	 *
+	 * For DMG exclude:
+	 * Do not change the pool map if the `pw_rf` is broken or is about to break,
+	 * unless the force option is given.
 	 */
-	if (src == MUS_SWIM && opc == MAP_EXCLUDE) {
+	if (!skip_rf_check) {
```

Comment:
this block of code seems only need to execute if "opc == MAP_EXCLUDE"?

Reply:
No, admin could force to skip this check...that is what this PR intent to do.

Reply:
the "--force" option is added to "dmg pool exclued" right? so it is still only for opc == MAP_EXCLUDE, I mean "if (!skip_rf_check && opc == MAP_EXCLUDE)", for other operation like EXTEND/REINT/DRAIN all need not to execute the code block?

Reply:
Fixed.

Reply:
Thx, not really a problem, I checked the code you added explicit flag parameter and set it as false for DRAIN/REINT/EXTEND. But not harmful to add the opc check.